pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -84,6 +85,49 @@ func (cs *ClusterStore) GetPilotClusters() []*k8s_cr.Cluster {
 	return cs.clusters
 }
 
+// ReadClustersV2 reads multiple clusters based upon the label istio/multiCluster
+func ReadClustersV2(k8s kubernetes.Interface, cs *ClusterStore) (errList error) {
+	err := getClustersConfigsV2(k8s, cs)
+	if err != nil {
+		// Errors were encountered, but cluster store was populated
+		log.Errorf("The following errors were encountered during multicluster label processing: [ %v ]",
+			err)
+	}
+
+	return nil
+}
+
+// getClustersConfigsV2 reads mutiple clusters from secrets with labels
+func getClustersConfigsV2(k8s kubernetes.Interface, cs *ClusterStore) (errList error) {
+	podNameSpace := os.Getenv("POD_NAMESPACE")
```

Comment:
NewServer in server.go does this same OS call to get the namespace just before calling initClusterRegistries.   So you should definitely pass the arg along rather than do another OS call here to get it.   Lot less expensive.

Reply:
good call.  Code modified.